File: MATTHEW/leetcode/AIcoding/5.Qwen-3/python3_solutions/1521-find-a-value-of-a-mysterious-function-closest-to-target.py
class Solution:
    def closestToTarget(self, A: List[int], target: int) -> int:
        s = {A[0]}
        res = abs(A[0] - target)
        for i in range(1, len(A)):
            t = set()
            t.add(A[i])
            for x in s:
                t.add(x & A[i])
            s = t
            for x in s:
                res = min(res, abs(x - target))
        return res

File: leetcode/AIcoding/5.Qwen-3/java_solutions/1521-find-a-value-of-a-mysterious-function-closest-to-target.java
public class Solution {

import java.util.*;

public class Solution {
    public int closestToTarget(int[] arr, int target) {
        Set<Integer> set = new HashSet<>();
        int res = Integer.MAX_VALUE;
        for (int num : arr) {
            Set<Integer> newSet = new HashSet<>();
            newSet.add(num);
            for (int s : set) {
                newSet.add(s & num);
            }
            set = newSet;
            for (int s : set) {
                res = Math.min(res, Math.abs(s - target));
            }
        }
        return res;
    }
}
}